### PR TITLE
Include component specific info in Renovate config

### PR DESCRIPTION
**Pull request by gnaponie**

### CODE REVIEW

**Comment by gnaponie** [comment]
@qixiang @FernandesMF @nvtnlucie PTAL

File: internal/pkg/component/gitlab/gitlab.go [hunk 1]
```diff
@@ -165,3 +172,34 @@ func (c *Component) getDefaultBranch() (string, error) {
...
+	baseConfig, err := c.GetRenovateBaseConfig(c.client, c.ctx)
+	if err != nil {
+		return "", err
+	}
+
+	baseConfig["platform"] = c.Platform
+	baseConfig["endpoint"] = c.GetAPIEndpoint()
+	// We don't need to set a username or gitAuthor for gitlab, since this is tight to a token
+	baseConfig["username"] = ""
+	baseConfig["gitAuthor"] = ""
```
Comment (qixiang) [inline internal/pkg/component/gitlab/gitlab.go:186]: I'm a little unsure about `gitAuthor` for gitlab repositories, maybe it's required, but we can update later after testing.

**Review by FernandesMF — APPROVED** [review]
Makes sense and looks good to me!

**Review by qixiang — APPROVED** [review]
LGTM

File: internal/pkg/component/gitlab/gitlab.go (hunk 1 above)
Comment (gnaponie) [reply internal/pkg/component/gitlab/gitlab.go:186]: Agreed, let's check during the testing phase. I'll add a note in the card. And merge this PR.
Thank you both for the review.

**Pull request merged**